Objects/lazyimportobject.c

```diff
@@ -0,0 +1,186 @@
+/* Lazy object implementation */
+
+#include "Python.h"
+#include "pycore_import.h"
+#include "pycore_lazyimportobject.h"
+#include "pycore_frame.h"
+#include "pycore_ceval.h"
+#include "pycore_interpframe.h"
+
+PyObject *
+_PyLazyImport_New(PyObject *builtins, PyObject *from, PyObject *attr)
+{
+    PyLazyImportObject *m;
+    if (!from || !PyUnicode_Check(from)) {
+        PyErr_BadArgument();
+        return NULL;
+    }
+    if (attr == Py_None) {
+        attr = NULL;
+    }
+    assert(!attr || PyObject_IsTrue(attr));
+    m = PyObject_GC_New(PyLazyImportObject, &PyLazyImport_Type);
+    if (m == NULL) {
+        return NULL;
+    }
+    Py_XINCREF(builtins);
+    m->lz_builtins = builtins;
+    Py_INCREF(from);
+    m->lz_from = from;
+    Py_XINCREF(attr);
+    m->lz_attr = attr;
```

Comment:
Wouldn't it be easier to use `Py_XNewRef`?
```suggestion
    m->lz_builtins = Py_XNewRef(builtins);
    m->lz_from = Py_XNewRef(from);
    m->lz_attr = Py_XNewRef(attr);
```